app/models/publisher.rb

```diff
@@ -48,39 +48,28 @@ def save_catching_gds_api_errors(&block)
         if content_model.save
           block.call
 
-          PublicationResponse.new(success: true)
+          Response.new(success: true)
         else
-          PublicationResponse.new(success: false)
+          Response.new(success: false)
         end
       end
     rescue GdsApi::HTTPErrorResponse => e
-      PublicationResponse.new(success: false, errors: e.error_details['error']['message'])
+      Response.new(success: false, error: e.error_details['error']['message'])
     end
   end
 
-  class DiscardDraftResponse
+  class Response
     def initialize(opts = {})
       @success = opts.fetch(:success)
-      @errors = opts.fetch(:errors, [])
+      @error = opts.fetch(:error, nil)
```

Comment:
Having an array allows us to return multiple errors without managing the rendering of them in this object. Say we want to render the errors in `<divs>` with formatting. If returning a string then we have to start joining the errors together here and inserting CSS classes. With an array we can leave that stuff in the view bit.

Empty arrays are nicer to deal with than nils too. Nils can be a pain.